packages/blade/src/components/Tooltip/_decisions/decisions.md

```diff
@@ -0,0 +1,199 @@
+
+1. Expose interaction props in all the trigger components
+
+- Web:
+  - onBlur
+  - onFocus
+  - onMouseLeave
+  - onMouseMove
+  - onPointerDown
+  - onPointerEnter
+- Native:
+  - onTouchStart
+  - onTouchEnd
+
+2. Expose the actual DOM node from ref instead of only exposing [certain methods via our useBladeInnerRef](https://github.com/razorpay/blade/blob/69a1bcef2f09ceaf6f910eaaca3076055fb059a2/packages/blade/src/hooks/useBladeInnerRef.web.ts#L26-L27) hook which we used to prevent component styling misuses. We need to expose the actual DOM node because FloatingUI internally does this [isElement() check](https://github.com/floating-ui/floating-ui/blob/b8990250568043b876e1c8fe42358fe337847ede/packages/react/src/hooks/useFloating.ts#L59) on the tooltip trigger element.
+
+</details>
+
+## Motion
+
+TBD
+
+## Accessibility
+
+- Tooltip will have [`role=tooltip`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role)
+- `Esc` should close the tooltip
+- When using keyboard tooltip will apear when target element is focused
+- Tooltip's must not contain interactive elements
+
+Resources:
+
+- https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/
+- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role
+- https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html
+- https://techservicesillinois.github.io/accessibility/aria-examples/tooltip.html
+- https://dequeuniversity.com/library/aria/tooltip
+
+## Open Questions
+
+<details>
+  <summary>Q1: Which components should and should not be qualify as trigger?</summary>
+
+Discussed with designers, we will allow all elements interactive, non-interactive to be used as triggers (eg, badge, counter, icons) because there can be many genuine usecases which we can't predict.
+
+</details>
+
+<details>
+  <summary>Q2: Mobile UX</summary>
+
+Should tooltips on mobile have open delay? eg: long press to open?
+
+There are certain cases where long press makes sense and certain places where immediate.
+
+**Long press example:**
+
+Google interactive icons, a search icon button might be clickable which shows a search input on clicked, but on long pressing it shows a tooltip.
+
+https://github.com/razorpay/blade/assets/35374649/51b8db65-7c46-4d7b-b388-61ab4fae8727
+
+**Immediately press (Razorpay icon tooltip):**
+
+Immediately shows tooltip on clicking on the icon for more info.
+
+https://github.com/razorpay/blade/assets/35374649/612a96b9-34e1-492b-8e81-e04b896d79db
+
+Immediate tooltips makes sense when the tooltip trigger doesn't do any extra action on clicking, for example imagine in that search icon button if an immediate tooltip is used the tooltip will never have a chance to show up since the screen changes and it opens a new dialog/page.
+
+**Auto dismissal:**
+
+Another UX we need to think about is,
+
+In the long press example, once you release the press the tooltip automatically dismisses after 1s.
```

Comment:
1s is too long? 300ms is what I've generally seen as a pattern most systems follow?

Reply:
This was added before RK finalized the duration will update this.